test/metadata-retrieval/metadataretrieval.go

```diff
@@ -0,0 +1,337 @@
+func (t *Test) Prepare() error {
+	return t.prepareMetadataRetrieval()
+}
+
+func (t *Test) prepareMetadataRetrieval() error {
+	t.log.Infof("Preparing metadata-retrieval binaries")
+	releases := regression.NewReleases("src-d", "metadata-retrieval", t.config.GitHubToken)
+
+	t.metadataRetrieval = make(map[string]*regression.Binary, len(t.config.Versions))
+	for _, version := range t.config.Versions {
+		b := NewMetadataRetrieval(t.config, version, releases)
+		err := b.Download()
+		if err != nil {
+			return err
+		}
+
+		t.metadataRetrieval[version] = b
+	}
+
+	return nil
+}
+
+// RunLoad executes the tests
+func (t *Test) RunLoad() error {
+	results := make(versionResults)
+
+	for _, version := range t.config.Versions {
+		_, ok := results[version]
+		if !ok {
+			results[version] = make(metadataRetrievalResults)
+		}
+
+		metadataRetrieval, ok := t.metadataRetrieval[version]
+		if !ok {
+			panic("metadataRetrieval not initialized. Was Prepare called?")
+		}
+
+		l := t.log.New(log.Fields{"version": version})
+
+		l.Infof("Running version tests")
+
+		times := t.config.Repeat
+		if times < 1 {
+			times = 1
+		}
+
+		// TODO(kyrcha): add a regression.yml file to metadata-retrieval
+		t.organizations = []string{"git-fixtures"}
+
+		for _, orgs := range t.organizations {
+			results[version][orgs] = make([]*Result, times)
+			for i := 0; i < times; i++ {
+				l.New(log.Fields{
+					"orgs": orgs,
+				}).Infof("Running query")
+
+				result, err := t.runLoadTest(metadataRetrieval, orgs)
+				results[version][orgs][i] = result
+
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	t.results = results
+
+	return nil
+}
+
+// runLoadTest runs metadata-retrieval download command and saves execution time + memory usage
+func (t *Test) runLoadTest(
+	metadataRetrieval *regression.Binary,
+	orgs string,
+) (*Result, error) {
+	t.log.Infof("Executing metadata-retrieval test")
+
+	command := NewCommand(metadataRetrieval.Path, orgs)
+	start := time.Now()
+	if err := command.Run(map[string]string{
+		"LOG_LEVEL":     "debug",
+		"GITHUB_TOKENS": t.config.GitHubToken,
+	}); err != nil {
+		t.log.With(log.Fields{
+			"orgs":               orgs,
+			"metadata-retrieval": metadataRetrieval.Path,
+		}).Errorf(err, "Could not execute metadata-retrieval")
+		return nil, err
+	}
+	wall := time.Since(start)
+	rusage := command.Rusage()
+
+	t.log.With(log.Fields{
+		"wall":   wall,
+		"memory": rusage.Maxrss,
+	}).Infof("Finished queries")
+
+	result := &regression.Result{
+		Wtime:  wall,
+		Stime:  time.Duration(rusage.Stime.Nano()),
+		Utime:  time.Duration(rusage.Utime.Nano()),
+		Memory: rusage.Maxrss * 1024,
+	}
+
+	r := &Result{
+		Result:        result,
+		Organizations: orgs,
+	}
+
+	return r, nil
+}
+
+// PrintTabbedResults prints table with results to stdout
+// Example:
+// Org                 | remote:add-regression-config
+// bblfsh,git-fixtures | 1m12.223594627s
+func (t *Test) PrintTabbedResults() {
```

Comment:
this and a couple of other implementations are almost equal here and in `gitcollector`, please add an unassigned `TODO` to refactor this part, we will take care of it later

Reply:
Added a TODO and a card to the private board